core/pta/rockchip/rk_secure_boot.c

```diff
@@ -0,0 +1,406 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2025, Pengutronix, Michael Tretter <entwicklung@pengutronix.de>
+ */
+
+#include <config.h>
+#include <drivers/rockchip_otp.h>
+#include <kernel/pseudo_ta.h>
+#include <kernel/tee_misc.h>
+#include <tee/uuid.h>
+#include <utee_defines.h>
+#include <stdio.h>
+#include <string.h>
+#include <platform_config.h>
+
+#include <pta_rk_secure_boot.h>
+
+#define PTA_NAME "rk_secure_boot.pta"
+
+/*
+ * Disable simulation to actually fusing the hash into the OTP. Enabling this
+ * option is necessary to actually enable secure boot with PTA, but may
+ * potentially brick your device.
+ */
+static const int simulation = IS_ENABLED2(CFG_RK_SECURE_BOOT_SIMULATION);
+
+static inline bool bit_test(uint32_t value, uint32_t bit)
+{
+	return (value & bit) == bit;
+}
+
+/* Convert hash from 32 bit words in OTP into a byte array */
+static TEE_Result otp_to_bytes(uint32_t *otp, size_t otp_size,
+			       uint8_t *bytes, size_t bytes_size)
+{
+	uint32_t *u32 = otp;
+	uint8_t *u8 = bytes;
+	size_t i;
+	size_t j;
+
+	if (bytes_size * sizeof(*bytes) != otp_size * sizeof(*otp))
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	for (i = 0; i < otp_size; i++, u32++)
```

Comment:
OP-TEE is very likely to always be little endian, especially where this PTA is compiled. How about adding a static_assert for that somewhere instead and use memcpy() directly?

Same for bytes_to_otp() below.